File: 003FizzBuzz/Go/fizzbuzz.go

package main

import "fmt"

const (
    FirstValue int = 1
    LastValue int = 100
)

func main(){

    fizzbuzz();

}


func fizzbuzz(){

    for i:=FirstValue; i<=LastValue; i++ {
        switch {
        case i%15==0: fmt.Println("FizzBuzz")
        case i%3==0: fmt.Println("Fizz");
        case i%5==0: fmt.Println("Buzz");
        default: fmt.Println(i)
        }
    }
}

File: 003FizzBuzz/CSharp/fizzbuzz.cs

using System;
using System.Linq;
using System.Collections.Generic;

class FizzBuzz
{

    const int FirstValue = 1;
    const int LastValue = 100;

    static void Main()
    {
        fizzbuzz();
    }

    static void fizzbuzz()
    {
        IEnumerable<int> num_list = Enumerable.Range(FirstValue, LastValue);
        bool mod3, mod5;

        foreach(int num in num_list)
        {
            mod3 = num%3 == 0;
            mod5 = num%5 == 0;

            if(mod3 && mod5)
            {
                Console.WriteLine("FizzBuzz");
            }
            else if(mod3)
            {
                Console.WriteLine("Fizz");
            }
            else if(mod5)
            {
                Console.WriteLine("Buzz");
            }
            else
            {
                Console.WriteLine(num);
            }
        }
    
    }
}
